packages/report-server/src/reportBuilder/transform/functions/fetchData/query/QueryBuilder.ts

```diff
@@ -0,0 +1,45 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2021 Beyond Essential Systems Pty Ltd
+ */
+
+import { Context } from '../../../../context';
+import { FetchReportQuery } from '../../../../../types';
+import { buildPeriodParams } from './buildPeriodParams';
+import { buildOrganisationUnitParams } from './buildOrganisationUnitParams';
+import { FetchConfig, ParsedFetchConfig } from '../types';
+import { TransformParser } from '../../../parser';
+import { TransformTable } from '../../../table';
+
+export class QueryBuilder {
+  private readonly parser: TransformParser;
+  private readonly ctx: Context;
+  private readonly config: FetchConfig;
+  private readonly query: FetchReportQuery;
+
+  public constructor(table: TransformTable, context: Context, config: FetchConfig) {
+    this.ctx = context;
+    this.config = config;
+    this.query = context?.request?.query;
+    this.parser = new TransformParser(table, this.ctx);
+  }
+
+  private parseConfig() {
+    const parsedConfig = { ...this.config };
+    Object.entries(parsedConfig).forEach(([field, value]) => {
+      if (typeof value === 'string') {
+        parsedConfig[field as keyof FetchConfig] = this.parser.evaluate(value);
+      } else {
+        parsedConfig[field as keyof FetchConfig] = value as any;
+      }
+    });
+    return parsedConfig as ParsedFetchConfig;
```

Comment:
Hmmm it's not a bad idea. But I worry it might be hard to avoid it being too strict... Maybe lets play with it in practice and if it turns out users could do with more error handling for incorrect data formats we can add a validation check here?